### PR TITLE
[CPU] Use elements count in the special nspc concat implementation

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *After some debugging, seems the RSS increase is related with posix_memalign() API reuse memory issue. Current is WA to not call the getMemory().getSize() interface to copy construct temp dnnl::memory::desc. 
 The debugging root cause detail is  https://jira.devtools.intel.com/browse/CVS-148778?focusedId=24976288&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24976288*

### Tickets:
 - *CVS-148778,CVS-144622*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/concat.cpp [hunk 1]
```diff
@@ -559,8 +559,8 @@ void Concat::execNspcSpecCase() {
 
         nonZeroInShapes++;
     }
-
-    const size_t iter_count = getParentEdgeAt(firstNonZeroEdge)->getMemory().getSize() / channelsDataSize[0];
+    const Shape& shape = getParentEdgeAt(firstNonZeroEdge)->getMemoryPtr()->getShape();
```
Comment (usstq) [inline src/plugins/intel_cpu/src/nodes/concat.cpp:562]: can firstNonZeroEdge be -1 here?

**Review by usstq — APPROVED** [review]
So there is no chance to fully solve it?

**Comment by luweizhou2016** [comment]
> So there is no chance to fully solve it?

It is a malloc limitation when allocating memory

> So there is no chance to fully solve it?

It seems to be  a posix_memalign() 

> So there is no chance to fully solve it?

1, It should be a limitation on posix_memalign() .. It is hard to avoid using posix_memalign()  this if we want to allocated cache line aligned memory.
2,  This limitation should only be triggered on some limited cases, concat patten with nhwc should exist in many models.

Based on this,  maybe cases by case handling and keep tracking whether similar issue arise in wide range in the future.

**Comment by dmitry-gorokhov** [comment]
@maxnick Please take a look on this

File: src/plugins/intel_cpu/src/nodes/concat.cpp (hunk 1 above)
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/concat.cpp:562]:
```suggestion
    const Shape& shape = getSrcMemoryAtPort(firstNonZeroEdge)->getShape();
```

**Review by maxnick — APPROVED** [review]

**Comment by maxnick** [comment]
> > So there is no chance to fully solve it?
> 
> 1, It should be a limitation on posix_memalign() .. It is hard to avoid using posix_memalign() this if we want to allocated cache line aligned memory. 2, This limitation should only be triggered on some limited cases, concat patten with nhwc should exist in many models.
> 
> Based on this, maybe cases by case handling and keep tracking whether similar issue arise in wide range in the future.

Please take a look at this PR https://github.com/openvinotoolkit/openvino/pull/26757. The thing is that we don't even need to copy the descriptor anymore, as the offsetPadding parameter is deprecated and in fact is not used.

**Comment by dmitry-gorokhov** [comment]
@luweizhou2016 Could you please confirm that PR mentioned by Maxim (https://github.com/openvinotoolkit/openvino/pull/26757) solves the issue? If yes I would prefer to close this one.

**Comment by luweizhou2016** [comment]
> @luweizhou2016 Could you please confirm that PR mentioned by Maxim (#26757) solves the issue? If yes I would prefer to close this one.

@dmitry-gorokhov , during my  previous debugging process, remove the tmp dnnl::memory::desc copy constructor also worked for RSS issue.  But  this PR can also merged in my understanding for performance consideration. shape should be lightweight in my understanding than descriptor invocation chain. how do you think?

**Comment by dmitry-gorokhov** [comment]
> > @luweizhou2016 Could you please confirm that PR mentioned by Maxim (#26757) solves the issue? If yes I would prefer to close this one.
> 
> @dmitry-gorokhov , during my previous debugging process, remove the tmp dnnl::memory::desc copy constructor also worked for RSS issue as Maksim PR did. But this PR can also merged in my understanding for performance consideration. shape should be lightweight in my understanding than descriptor invocation chain. how do you think?

Not sure if the perf difference is high enough to affect e2e op perf, but I don't mind. Lets merge then.

**Pull request merged**